tests/test_tensor/test_module_spec.py

```diff
@@ -110,28 +122,79 @@ def run_linear_with_spec(label):
     assert tensor_shard_equal(model.weight.grad, model_handy.weight.grad)
     assert tensor_shard_equal(model.bias.grad, model_handy.bias.grad)
 
+def run_check_duplicated_param():
```

Comment:
Do you mean sharded param? you'd better use a consistent concept.
Duplicated means the param has two copies. shared is only one copy but pointed by two ptr.

Reply:
Sry. I just mean shared param.

Reply:
Updated